src/custom-element.js

```diff
@@ -637,13 +641,9 @@ function createBaseCustomElementClass(win) {
       this.assertLayout_();
       this.implementation_.layout_ = this.layout_;
       this.implementation_.layoutWidth_ = this.layoutWidth_;
-      if (this.everAttached) {
-        this.implementation_.firstAttachedCallback();
```

Comment:
These two lines were duplicated in `#connectedCallback` [here](https://github.com/ampproject/amphtml/pull/8229/files#diff-8e1cdba946c07a7e631ef74645a5d6f0L948). But they're only called if upgrade happens sync, and these two lines are only called if upgrade happens async.

Because `#completeUpgrade` is already called sync or async appropriately, we can just move the lines here and always call them.